source/MaterialXView/ShaderValidators/Glsl/GlslValidator.h

```diff
@@ -0,0 +1,326 @@
+#ifndef MATERIALX_GLSLVALIDATOR_H
```

Comment:
Main class which has all the logic in it for now. Will look at splitting this up and simplifying the interface slowly. As mentioned the proposal is to separate out a GLSL-program with creation and validation. Input binding and checks and rendering. The validation should eventually have a simple interface as you suggested of:
1) setup using shader strings or from a HwShader
2) render (optionally save to disk)